# Math 134A Week 5

# Short Rates
Given the spot rates $(s_1, s_2,\ldots)$. We define the **short rates** at time $k$ as $r_k = f_{k, k+1}$, and $r_0 = s_1$. The short rates $r_k$ is the interest rate for the 1-year term loan at time $k$.

Then we have the equation 
$$(1+s_k)^k = (1+r_0)\cdots(1+r_k)$$

$$(1+f_{i,j})^{j-1} = (1+r_i)\cdots(1+r_j)$$

Given the short rates, we can also determine the spot rates uniquely.


# Invariance Theorem
**Theorem** \
We suppose that the interest rate evolve under the **expectation dynamics** (i.e. the spot rates of next year are given by the forward rates $s_1' = f_{1, 2},\;s_2' = f_{1,3},\ldots$), and the interest compounds annually. Then, a sum of money invested in the interest rate market for n years will grow by a factor of $(1+s_n)^n$, independent of the investment and reinvestment strategy as long as all funds are fully invested. 

**Example** \
Suppose that the spot rates are $(0.1, 0.2)$. We consider an investment of 100 dollars in 2 years. One option is that we may purchase the 1-year certificate of deposit, and we will have 110 at year 1. By the expectation dynamic, the new spot rate for 1 year investment is $f_{1,2} = 1.2^2/1.1 - 1$. If we reinvest the 110 at this interest rate, we get $110 \cdot (1.2^2/1.1) = 144$ at year 2. The other option is that we can purchase the 2-year certificate of deposit, and we will get $100 \cdot 1.2^2 = 144$ at year 2. Those two investments provide the same return at the end of year 2. 




# Running Present Value
We consider a cashflow $(x_0, x_1,\ldots, x_n)$, where $x_i$ is received in year $i$. Suppose that the spot rates are $(s_1,s_2,\ldots, s_n)$, and we define the discount factor $d_k = d_{0, k} = \frac{1}{(1+s_k)^k}$, which is the discount factor from time k to time 0. We may also define the discount factor $d_{j,k} = \frac{1}{(1+f_{j, k})^{k-j}}$, which is the discount factor from time $k$ to time $j$. Let $PV(0)$ be the present value of this cashflow at time $0$, then we have 
$$PV(0) = x_0+x_1d_1+\cdots+x_kd_k$$

Let $PV(k)$ to be the present value of the remaining cashflow at time k. In other words, $PV(k)$ is the present value of the cashflow $(x_k, x_{k+1},\ldots, x_n)$ under the spot rates at time $k$, which is $(f_{k, k+1}, f_{k, k+2},\ldots, f_{k, k+n})$. Then we have 
$$PV(k) = x_k+x_{k+1}d_{k, k+1}+\cdots+x_{n}d_{k, n}$$

Then, we have the following formula for computing $PV(0)$ recursively
$$\begin{cases}
PV(n) = x_n \\
PV(k) = x_k+d_{k, k+1}P(k+1)
\end{cases}$$







**Exercise 4.11** (Running PV example) \
A (yearly) cash flow stream is $x = (—40, 10,10, 10, 10, 10, 10)$. The spot rates are (5.0, 5.3, 5.6, 5.8, 6.0, 6.1) \
(a) Find the current discount factors $d_{0, k}$, and use them to determine the (net) present value of the stream. \
(b) Find the series of expectations dynamics short-rate discount factors, and use the running present value method to evaluate the stream.

In [17]:
import pandas as pd

cashflow = [-40, 10, 10, 10, 10, 10, 10]
spot_rates = [5, 5.3, 5.6, 5.8, 6, 6.1]
discount_factors_0 = []
for k in range(len(spot_rates)):
    discount_factors_0.append(1/ (1 + spot_rates[k]/100)**(k+1))
discount_factors_0_formatted = [ '%.3f' % elem for elem in discount_factors_0]
print("a) The current discount rates d_{0,k} are ", discount_factors_0_formatted)
print("   We use the following table to compute the present value of this cashflow")

table_npv_methods = pd.DataFrame(index=['', '', '', ' ', ' ', ' ', ' '])

table_npv_methods['Year'] = [0, 1, 2, 3, 4, 5, 6]
table_npv_methods['Cashflow'] = cashflow 
table_npv_methods['Spot'] = [''] + spot_rates
table_npv_methods['Discount Factor'] = [1] + discount_factors_0
table_npv_methods['Present Value'] = table_npv_methods['Cashflow'] * table_npv_methods['Discount Factor']
npv = sum(table_npv_methods['Present Value'])
table_npv_methods.loc['Sum'] = ['', ' ',  ' ', ' ', npv] 

display(table_npv_methods)
print(f"The net present value of this cashflow is {npv:.3f}\n")

print("b) If we compute the present value by the running PV methods, then we have the following")

print(f"PV({6}) = {cashflow[-1]:.2f}")
pv_last_step = cashflow[-1]
for k in reversed(range(0, len(cashflow) - 1)):
    discount_factor = (1 + spot_rates[k-1]/100)**k / (1 + spot_rates[k]/100) ** (k+1)
    new_pv = cashflow[k] + pv_last_step * discount_factor
    print(f"PV({k}) = {cashflow[k]:.2f} + {pv_last_step:.3f} * {discount_factor:.4f} = {new_pv:.3f}")
    pv_last_step = new_pv

a) The current discount rates d_{0,k} are  ['0.952', '0.902', '0.849', '0.798', '0.747', '0.701']
   We use the following table to compute the present value of this cashflow


Year Cashflow Spot Discount Factor  Present Value
       0      -40                  1.0     -40.000000
       1       10    5        0.952381       9.523810
       2       10  5.3        0.901869       9.018686
       3       10  5.6        0.849197       8.491966
       4       10  5.8          0.7981       7.981000
       5       10    6        0.747258       7.472582
       6       10  6.1        0.700983       7.009833
Sum                                          9.497877

The net present value of this cashflow is 9.498

b) If we compute the present value by the running PV methods, then we have the following
PV(6) = 10.00
PV(5) = 10.00 + 10.000 * 0.9381 = 19.381
PV(4) = 10.00 + 19.381 * 0.9363 = 28.146
PV(3) = 10.00 + 28.146 * 0.9398 = 36.453
PV(2) = 10.00 + 36.453 * 0.9416 = 44.324
PV(1) = 10.00 + 44.324 * 0.9470 = 51.973
PV(0) = -40.00 + 51.973 * 0.9524 = 9.498


# Duration and Sensitivity to Parallel Shift of the Spot Curve
### Fisher-Weil Duration
We define the duration for a cashflow in a similar way under the term structure. Under the continuous compounding model, we consider a spot rate curve given by $s_t$, where the discount factor payment received in year $t$ is $e^{-ts_t}$. The **Fisher-Weil Duration** of a cashflow $(x_{t_0},\ldots, x_{t_n})$ is defined by 
$$D_{FW} = \frac{1}{PV}\sum_0^n t_kx_{t_k}e^{-s_{t_k}t_k}$$ 
This is exactly the weighted sum of the time we receive each payment, and the weights are the ratio if the present value of each payment over the total present value of the cashflow. 

We call $s_t+\lambda$ as a **parallel shift** of the spot rate curve. In other words, we are adding $\lambda$ to all spot rates. We may represent the present value as a function of $\lambda$ as 
$$P(\lambda) = \sum_0^n x_{t_k}e^{-t_k(s_{t_k}+\lambda)}$$
Then, we have 
$$\frac{dP(\lambda)}{d\lambda}\Big|_{\lambda = 0} = -D_{FW}/P(0)$$

In other words, the Fisher-Weil duration measures the sensitivity of the price relative to parallel shift in the spot rates.

### Quasi-modified duration
Suppose that the interest compounds $m$ times every year. Then we may define the **Quasi-modified duration** in a similar way, as the sensitivity to the parallel shift of the spot rates. Let $(x_0,\ldots, x_n)$ be a cashflow, where the payment $x_k$ is received in the period $k$. Let $(s_1,\ldots, s_n)$ be the spot rates of each period, quoted as annual interest. Then we have
$$D_M = -\frac{1}{P(0)}\frac{dP(\lambda)}{d\lambda}\Big|_{\lambda = 0} = \frac{1}{PV}\sum_1^n \left(\frac{k}{m}\right)x_k\left(1+\frac{s_k}{m}\right)^{-(k+1)}$$

Note that this is also a weighted sum of the time that we receive each payment. The weights are almost the ratio of the present value of each payment over the present value of the total cashflow, except that we multiplied $(1+\frac{s_k}{m})^{-(k+1)}$ instead of $(1+\frac{s_k}{m})^{-k}$ for discounting. 

**Example**
Suppose that the spot rates are given by $(7.67, 8.27, 8.81, 9.31, 9.75, 10.16, 10.52, 10.85, 11.15, 11.42, 11.67, 11.89)$. We consider two bonds, where bond A is a 12-year 6\% bond, and bond B is a 5-year 10% bond. Assume that the coupon are paid annually. Compute price and the quasi-modified duration of the above two bonds. 

In [46]:
import pandas as pd
spot_rates = [7.67, 8.27, 8.81, 9.31, 9.75, 10.16, 10.52, 10.85, 11.15, 11.42, 11.67, 11.89]
cashflow_A = [0] + [6] * 11 + [106]
cashflow_B = [0] + [10] * 4 + [110]
P_A = 65.95
P_B = 101.66

discount_factors = [1]
for k in range(len(spot_rates)):
    discount_factors.append(1/ (1 + spot_rates[k]/100)**(k+1))

modified_discount_factors = [1]
for k in range(len(spot_rates)):
    modified_discount_factors.append(1/ (1 + spot_rates[k]/100)**(k+2))

year = list(range(12+1))
table = pd.DataFrame(index = ['']*13)
table['year'] = year
table['Spot rates'] = ['']+spot_rates
table['Bond A'] = cashflow_A
table['Bond B'] = cashflow_B + [0] * 7
table['Discount Factor'] = discount_factors
table['PV A'] = table['Bond A'] * table['Discount Factor']
table['PV B'] = table['Bond B'] * table['Discount Factor']
table['Modified Discount Factor'] = modified_discount_factors
table['PV\' A'] = table['Bond A'] * table['Modified Discount Factor'] * table['year']
table['PV\' B'] = table['Bond B'] * table['Modified Discount Factor'] * table['year']

PV_A = sum(table['PV A'])
PV_B = sum(table['PV B'])
P_derivative_A = sum(table['PV\' A'])
P_derivative_B = sum(table['PV\' B'])

table.loc['Sum'] = ['', '','', '', '', PV_A, PV_B, '', P_derivative_A , P_derivative_B] 
table.loc['Duration'] = ['', '','', '', '', '', '', '', P_derivative_A/PV_A, P_derivative_B/PV_B] 

display(table)

year Spot rates Bond A Bond B Discount Factor       PV A        PV B  \
            0                 0      0             1.0        0.0         0.0   
            1       7.67      6     10        0.928764   5.572583    9.287638   
            2       8.27      6     10        0.853068   5.118409    8.530681   
            3       8.81      6     10        0.776236   4.657414    7.762356   
            4       9.31      6     10        0.700423   4.202538     7.00423   
            5       9.75      6    110        0.628026   3.768154   69.082817   
            6      10.16      6      0        0.559573   3.357436         0.0   
            7      10.52      6      0        0.496494   2.978963         0.0   
            8      10.85      6      0        0.438646   2.631877         0.0   
            9      11.15      6      0        0.386202   2.317213         0.0   
           10      11.42      6      0        0.339132   2.034791         0.0   
           11      11.67      6      0         0.29696   1.781762         0.0   
           12      11.89    106      0         0.25972  27.530277         0.0   
Sum                                                     65.951416  101.667724   
Duration                                                                        

         Modified Discount Factor       PV' A       PV' B  
                              1.0    0.000000    0.000000  
                         0.862602    5.175613    8.626022  
                         0.787908    9.454898   15.758163  
                         0.713386   12.840953   21.401589  
                         0.640768   15.378422   25.630703  
                         0.572233   17.166987  314.728097  
                         0.507964   18.286686    0.000000  
                         0.449234   18.867844    0.000000  
                         0.395712   18.994154    0.000000  
                          0.34746   18.762862    0.000000  
                         0.304372   18.262346    0.000000  
                         0.265927   17.551157    0.000000  
                          0.23212  295.257237    0.000000  
Sum                                465.999159  386.144574  
Duration                             7.065795    3.798104

The modified discount factor in the above table are computed by $(1+s_k)^{-(k+1)}$ as in the duration formula, and the PV\' A is $k\cdot x_k \cdot(1+s_k)^{-(k+1)}$, which is the negative of the derivative of $x_k(1+s_k+\lambda)^{-k}$ at $\lambda = 0$. The computation shows that bond A has price 65.95, and duration 7.07 years, and bond B has price 101.67, and duration 3.80 years. 

# Immunization
The Immunization under the term structure is exactly the same as the immunization we covered previously. We will match both the present value, and the duration of the portfolio with the given obligation. 


**Exercise 4.15**(Stream immunization) 
A company faces a stream of obligations over the next 8 years as shown: where the numbers denote thousands of dollars. The spot rate curve is that of the above table. Try to find a portfolio, consisting of the two bonds described in the above example, that has the same present value as the obligation stream and is immunized against an additive shift in the spot rate curve.

| year | 1   | 2   | 3   | 4   | 5   | 6   | 7   | 8  |
|------|-----|-----|-----|-----|-----|-----|-----|----|
|      | 500 | 900 | 600 | 500 | 100 | 100 | 100 | 50 |


**Solution**
We first compute the price $P$, and the duration $D$ of this obligation. Then we solve the system of linear equations 
$$\begin{cases}
x_1P_A+x_2P_B = P \\ 
x_1\frac{P_A}{P}D_A+x_2\frac{P_B}{P}D_B = D
\end{cases}$$ 
where $P_A, P_B$ are the price of bond A, bond B computed from above, and $D_A, D_B$ are the quasi-modified duration of bond $A$ and bond $B$. The solution $x_1$ and $x_2$ means a portfolio with $x_1$ units of bond A, and $x_2$ units of bond B. 

In [48]:
cashflow = [0, 500, 900, 600, 500, 100, 100, 100, 50]
pv = 0
pv_derivative = 0
for i in range(len(cashflow)):
    pv += cashflow[i] * discount_factors[i]
    pv_derivative += i * cashflow[i] * modified_discount_factors[i]
    
print(f"The present value of the obligation is {pv:.2f}")
print(f'The duration of this obligation is {pv_derivative/pv:.2f} years')

The present value of the obligation is 2238.44
The duration of this obligation is 2.45 years


Then We solve the following system of linear equations
$$\begin{cases}
65.95x_1 + 101.67x_2 = 2238.44 \\
\frac{65.95\cdot 7.07}{2238.44}x_1 + \frac{101.67\cdot 3.80}{2238.44}x_2 = 2.45
\end{cases}$$

We see that $x_1 = -13.84$, $x_2 = 31.00$

**Exercise 5.1**:
(Capital budgeting) A firm is considering funding several proposed projects that have the financial properties shown in the following table. The available budget is 600,000. What set of projects would be recommended by the approximate method based on benefit-cost ratios? What is the optimal set of projects?

| Project | Outlay (in 1,000) | Present Worth |
|---------|-------------------|---------------|
| 1       | 100               | 200           |
| 2       | 300               | 500           |
| 3       | 200               | 300           |
| 4       | 150               | 200           |
| 5       | 150               | 250           | 


In [5]:
from scipy.optimize import linprog
obj = [-200, -500, -300, -200, -250]
lhs_ineq = [[100, 300, 200, 150, 150]]
rhs_ineq = [600]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, integrality = [1,1,1,1,1], bounds = [(0,1)])
opt.x

array([1., 1., 1., 0., 0.])

Based on the benefit cost ratio approximation, it is optimal to select project 1,2,5. The net cost is 550, and NPV is 950
Based on the above computation, we see that it is most optimal to select project 1,2,3. The net cost is 600, and NPV is 1000

**Exercise 5.2** (County transportation choices) Suppose that the goals and specific projects shown in the table are being considered by the County Transportation Authority. The total available budget is 5 million.

|                                 | Cost (in 1000) | NPV (in 1000) |
|---------------------------------|----------------|---------------|
| - Road between Augen and Burger |                |               | 
| 1. Concrete, 2 lanes            | 2000           | 4000          |
| 2. Concrete, 4 lanes            | 3000           | 5000          |
| 3. Asphalt, 2 lanes             | 1500           | 3000          |
| 4. Asphalt, 4 lanes             | 2200           | 4300          |
| - Bridge at Cay Road            |                |               | 
| 5. Repair existing              | 500            | 1000          |
| 6. Add lane                     | 1500           | 1500          |
| 7. New Structure                | 2500           | 2500          |
| - Traffic Control in Downsberg  |                |               | 
| 8. Traffic lights               | 100            | 300           |
| 9. Turn Lanes                   | 600            | 1000          |
| 10. Underpass                   | 1000           | 2000          |

The bridge at Cay Road is actually part of the road between Augen and Burger. Therefore it is not reasonable for the bridge to have fewer lanes than the road itself. This means that if projects 2 or 4 are carried out, either projects 6 or 7 must also be carried out. Formulate a zero—one programming problem that includes this additional requirement. Solve the problem.                            

**Solution:**
Suppose $x_i\in\{0,1\}$ denotes either we choose project $i$ or not. Then, we would like to maximize 
$$4000x_1+5000x_2+\cdots+2000x_{10}$$
subject to the constraints
$$\begin{align}
2000x_2+3000x_2+\cdots + 1000x_{10} &\leq 5000 \\
x_1+x_2+x_3+x_4 &\leq 1 \\
x_5+x_6+x_7 &\leq 1 \\
x_8+x_9+x_{10} &\leq 1 \\
x_6+x_7-x_2-x_4 &\geq 0
\end{align}$$

Note that the last condition is saying that if $x_2+x_4 = 1$ (i.e, project 2 or 4 are carried out), then $x_6+x_7 = 1$ (i.e, project 6 or project 7 must be carried out)

We run the following program so solve this linear programming problem. We see that it is most optimal to select project 4, 6, 10. The net cost is 4700,000 and NPV is 7800,000

In [13]:
from scipy.optimize import linprog
obj = [-4000, -5000, -3000, -4300, -1000, -1500, -2500, -300, -1000, -2000]
lhs_ineq = [[2000, 3000, 1500, 2200, 500, 1500, 2500, 100, 600, 1000], [1,1,1,1,0,0,0,0,0,0], [0,0,0,0,1,1,1,0,0,0], [0,0,0,0,0,0,0,1,1,1], [0,1, 0, 1, 0, -1, -1, 0, 0, 0]]
rhs_ineq = [5000, 1, 1, 1, 0]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, integrality = [1,1,1,1,1,1,1,1,1,1], bounds = [(0,1)])
opt.x

array([0., 0., 0., 1., 0., 1., 0., 0., 0., 1.])

**Exercise 5.3**
(Two-period budget) A company has identified a number of promising projects, as indicated in Table 5.7. The cash flows for the first 2 years are shown (they are all negative). The cash flows in later years are positive, and the net present value of each project is shown. The company managers have decided that they can allocate up to 250,000 in each of the first 2 years to fund these projects. If less than 250,000 is used the first year, the balance can be invested at 10% and used to augment the next year’s budget. Which projects should be funded?

| project | 1   | 2    | NPV |
|---------|-----|------|-----|
| 1       | -90 | -58  | 150 |
| 2       | -80 | -80  | 200 |
| 3       | -50 | -100 | 100 |
| 4       | -20 | -64  | 100 |
| 5       | -40 | -50  | 120 |
| 6       | -80 | -20  | 150 |
| 7       | -80 | -100 | 240 |

**Solution:**
We first convert the problem to a linear programing problem. Suppose $x_i\in \{0,1\}$ denotes either we select project $i$ or not. Let $y\geq 0$ denote the surplus of the funds in year 1. Then, we would like to maximize 
$$150x_1+200x_2+\cdots+240x_7$$
subject to the constraints
$$90x_1+80x_2+\cdots+80x_7+y\leq 250$$
$$58x_1+80x_2+\cdots+240x_7 \leq 250 + 1.1y$$
We run the following program to solve this linear programming problem. We see that it is most optimal to select project 4,5,6,7

In [12]:
from scipy.optimize import linprog
from scipy.optimize import milp
obj = [-150, -200, -100, -100, -120, -150, -240, 0]
lhs_ineq = [[90, 80, 50, 20, 40, 80, 80, 1], [58, 80, 100, 54, 50, 20, 100, -1.1], [1,0,0,0,0,0,0,0], [0,1,0,0,0,0,0,0], [0,0,1,0,0,0,0,0], [0,0,0,1,0,0,0,0], [0,0,0,0,1,0,0,0], [0,0,0,0,0,1,0,0], [0,0,0,0,0,0,1,0]]
rhs_ineq = [250, 250, 1, 1, 1, 1, 1, 1, 1]
opt = linprog(c=obj, A_ub=lhs_ineq, b_ub=rhs_ineq, integrality = [1,1,1,1,1,1,1,0])
opt.x

array([0.       , 0.       , 0.       , 1.       , 1.       , 1.       ,
       1.       , 1.9047619])

**Exercise 5.4** (Bond matrix) The cash matching and other problems can be conveniently represented in matrix form. Suppose there are m bonds. We define for each bond j its associated yearly cash flow stream (column) vector $\vec{c_j}$, which is n-dimensional. The yearly obligations are likewise represented by the n-dimensional vector $\vec{y}$. We can stack the $c_j$ vectors side by side to form the columns of a bond matrix C. Finally we let $\vec{p}$ and $\vec{x}$ be m-dimensional column vectors. The cash matching problem can be expressed as
$$\text{minimize } \vec{p}^T\vec{x}$$
$$\text{subject to } \begin{align}&C\vec{x} > \vec{y} \\ &\vec{x}\geq 0\end{align}$$
(a) Identify $C,\vec{y},\vec{p}$ and $\vec{x}$ in table 5.3  \
(b) Show that if all bonds are priced according to a common term structure of interest rates, there is a vector $\vec{v}$ satisfying
$$C^T\vec{v}= \vec{p}$$
What are the components of $\vec{v}$? \
(c) Suppose $\vec{b}$ is a vector whose components represent obligations in each period. Show that a portfolio $\vec{x}$ meeting these obligations exactly satisfies $C\vec{x} =\vec{b}$. \
(d) With $\vec{x}$ and $\vec{v}$ defined as before, show that the price of the portfolio $\vec{x}$ is $\vec{v}^T\vec{b}$. Interpret this result.


**Solution** \
a) $C = \begin{bmatrix}10 & 7 & 8 & 6 & 7 & 5 & 10 & 8 & 7 & 100 \\
10 & 7 & 8 & 6 & 7 & 5 & 10 & 8 & 107 & 0 \\ 
10 & 7 & 8 & 6 & 7 & 5 & 110 & 108 & 0 & 0 \\
10 & 7 & 8 & 6 & 7 & 105 & 0 & 0 & 0 & 0 \\
 10 & 7 & 8 & 106 & 107 & 0 & 0 & 0 & 0 & 0 \\
  110 & 107 & 108 & 0 & 0 & 0 & 0 & 0 & 0 & 0\end{bmatrix}$, $\vec{y} = \begin{bmatrix}100 \\ 200 \\ 800 \\ 100 \\ 800 \\ 1200\end{bmatrix}, \vec{p} = \begin{bmatrix}109\\ 94.8 \\ 99.5 \\ 93.1 \\ 97.2 \\ 92.9 \\ 110 \\ 104 \\ 102 \\ 95.2\end{bmatrix}, \vec{x} = \begin{bmatrix}0 \\ 11.2 \\ 0 \\ 6.81 \\ 0 \\ 0 \\ 0 \\ 6.3 \\ 0.28 \\ 0\end{bmatrix}$
  
b) If bonds are priced according to term structure $(s_1, s_2,\ldots, s_n)$ where $s_i$ is spot rates at year $i$. Then we define the vector $\vec{v} = (d_1,\ldots, d_n)$ where $d_i = \frac{1}{(1+s_i)^i}$. Then the price of the bond $i$, which has cashflow $\vec{c_j}$ is computed by 
$$p_i = c_{i1}d_1+c_{i2}d_2+\cdots+c_{in}d_n = \vec{c_i}^T\vec{v}.$$
Hence, we have 
$$\vec{p} = C^T\vec{v}$$
The components of $\vec{v}$ are the discount factors $d_i$. 

c) This is directly from the definition. The $i$-th entry of $C\vec{x}$ is the cashflow of this portfolio at year $i$. 

d) The price of this portfolio is $\vec{x}^T\vec{p} = \vec{x}^T(C^T\vec{v}) = (C\vec{x})^T\vec{v} = \vec{b}^T\vec{v} = \vec{v}^T\vec{b}$. 

This result means that if the cashflow of this portfolio equals the obligation. Then the value of this portfolio is exactly the present value of the obligation. 


**Exercise 5.9** (Little Bear Oil) You have purchased a lease for the Little Bear Oil well. This well has initial reserves of 100 thousand barrels of oil. In any year you have three choices of how to operate the well: (a) you can not pump, in which case there is no operating cost and no change in oil reserves; (b) you can pump normally, in which case the operating cost is 50 thousand, and you will pump out 20% of what the reserves were at the beginning of the year; or (c) you can use enhanced pumping using water pressure, in which case the operating cost is 120 thousand, and you will pump out 36% of what the reserves were at the beginning of the year. The price of oil is 10 per barrel and the interest rate is 10%. Assume that both your operating costs and the oil revenues come at the beginning of the year (through advance sales). Your lease is for a period of 3 years.